CONTRIBUTING.md

```diff
@@ -360,11 +360,11 @@ where link local is being tested.
     used for traffic testing; split as needed.
 *   `BMWG`: (198.18.0.0/15): additional data plane networks.
 
-*   20.0.0.1/15: data plane source network addresses used for scale testing.
+*   20.0.0.0/8: data plane source network addresses used for scale testing.
```

Comment:
I spoke to @xw-g. Can we use different subnets to avoid potential interference with publicly allocated subnets?

  - 20.0.0.0/8 is allocated to Microsoft and others. Suggest to use 251.0.0.0/8 instead.
  - 30.0.0.0/8 is allocated to the Depart of Defense. Suggest to use 252.0.0.0/8 instead.
  - 100.0.0.0/8 is reserved for Carrier NAT and others. Suggest to use 253.0.0.0/8 instead.
  - If you need more, we can also use 254.0.0.0/8
  - Avoid 255.0.0.0/8 because 255.255.255.255/32 may have a special meaning.
 
The suggested range is reserved for future use since 1981. I'm hoping that even if they start using them, the allocation will start from 240.0.0/8.

https://www.iana.org/assignments/ipv4-address-space/ipv4-address-space.xhtml

Reply:
@liulk , @xw-g  I will update both pull requests with this new address range.

Reply:
> @liulk , @xw-g I will update both pull requests with this new address range.

@liulk , @xw-g , > 240.0.0/4 orlonger is Martian addresses ? routes will be not accepted? These ranges does not work.
please suggest anything which is lesser than this range?

Reply:
Is it reasonable to assume that tests that need extended IP address range for scale testing will need to customize the Martian range to allow for these extended addresses? Otherwise there really isn't any other spaces.

Another lesser evil is to find a /8 subnet allocated to a legacy company that no longer announces BGP prefixes, e.g. 19./8: https://bgp.he.net/ip/19.0.0.1 but this is the only one I could find.

Scaling tests really ought to be using IPv6. @dplore @xw-g to comment.

Reply:
Agreed that we should use private/test reserved IP ranges whenever possible.  Because these are open source tests, we expect them to be run in many environments in which properly allocated IP's might be present and cause a conflict.  

Let us try using 240.0.0.0/4.  It seems some devices need a special command to permit  martian addresses to be used: 
-  [Allowing Martian addresses on JunOS](https://www.juniper.net/documentation/us/en/software/junos/static-routing/topics/topic-map/recognize-martian-addr-routing.html#d32e54__d8789e135)
- [Martian addresses on Cisco IOS XR](https://www.cisco.com/c/en/us/td/docs/iosxr/cisco8000/bgp/73x/b-bgp-cg-8k-73x/implementing-bgp.html#concept_vym_phb_4jb)

Let's try using 240.0/4 for the scaling test.  If a blocking issue arises that we cannot address with a simple deviation, we will pick some other IPv4 block.

Regarding using IPv6 for scaling, I think we want to flexibility to scale both IPv4 and IPv6 routes.  This is desirable because hardware behaves differently for v4 and v6 especially in regards to scaling forwarding tables.

Reply:
I agree with @dplore's suggestion and comment. 

I would add that, re using an unannounced subnet (such as 19.0.0.0/8), the absence of a prefix in the "global" BGP table of any one (or even many) AS is not necessarily indicative of the prefix being unused. For example, I'm aware of some networks that accept the risk and use unannounced blocks internally for various purposes, e.g., CGN pools instead of the allocated 100.64 block, additional "private" space, etc. Further, some networks use globally unique addresses for private internets that aren't seen on the "global" table, e.g., securities trading exchanges.

That's not to say that we have no option if 240.0/4 doesn't work for us -- but using globally unannounced space *does* add some additional risk if people use these tests in less-than-perfectly controlled lab environments.

Reply:
I have updated with Martian address range and removed 20/30/100 address range .  We will update baseline config which is needed to allow Martian address range.